rpcs3/Emu/Io/usio.cpp

```diff
@@ -263,23 +262,23 @@ void usb_device_usio::translate_input()
 	std::memcpy(input_buf.data(), &test_keys, sizeof(u16));
 	std::memcpy(input_buf.data() + 16, &coin_counter, sizeof(u16));
 
-	q_replies.push(input_buf);
-	q_replies.push({0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00});
+	response = std::move(input_buf);
 }
 
-void usb_device_usio::usio_write(u8 channel, u16 reg, const std::vector<u8>& data)
+void usb_device_usio::usio_write(u8 channel, u16 reg, std::vector<u8>& data)
 {
-	auto write_memory = [&](std::vector<u8>& memory, bool exact_size = true)
+	auto write_memory = [&](std::vector<u8>& memory)
 	{
-		ensure(data.size() == memory.size() || (data.size() <= memory.size() && !exact_size));
-		memcpy(memory.data(), data.data(), data.size());
+		auto size = memory.size();
+		memory = std::move(data);
+		memory.resize(size);
```

Comment:
isn't this resize redundant?

Reply:
There could be rare chances that `size` is smaller than `memory.size()`, and the `resize` here is just truncating it.